modules/axhal/src/platform/pc_x86/acpi.rs

```diff
@@ -0,0 +1,354 @@
+extern crate alloc;
+
+use alloc::boxed::Box;
+use alloc::format;
+use core::alloc::{AllocError, Layout};
+use core::ptr::NonNull;
+
+use acpi::{AcpiTables, PhysicalMapping};
+use aml::pci_routing::{IrqDescriptor, PciRoutingTable, Pin};
+use aml::{AmlContext, AmlName, DebugVerbosity};
+
+use crate::mem::phys_to_virt;
+use axalloc::global_allocator;
+use lazy_init::LazyInit;
+use memory_addr::PhysAddr;
+
+use crate::arch::irq_to_vector;
+
+#[derive(Clone)]
+struct LocalAcpiHandler;
+
+impl acpi::AcpiHandler for LocalAcpiHandler {
+    unsafe fn map_physical_region<T>(
+        &self,
+        physical_address: usize,
+        size: usize,
+    ) -> PhysicalMapping<Self, T> {
+        let vaddr = phys_to_virt(PhysAddr::from(physical_address)).as_usize();
+        PhysicalMapping::new(
+            physical_address,
+            NonNull::new_unchecked(vaddr as *mut i32 as *mut _),
+            size,
+            size,
+            self.clone(),
+        )
+    }
+    fn unmap_physical_region<T>(_region: &PhysicalMapping<Self, T>) {}
+}
+
+struct LocalAmlHandler;
+
+impl aml::Handler for LocalAmlHandler {
+    fn read_u8(&self, address: usize) -> u8 {
+        let vaddr = phys_to_virt(PhysAddr::from(address)).as_ptr();
+        unsafe { vaddr.read_volatile() }
+    }
+
+    fn read_u16(&self, address: usize) -> u16 {
+        let vaddr = phys_to_virt(PhysAddr::from(address)).as_ptr() as *const u16;
```

Comment:
Use `as_ptr::<u16>()`

Reply:
Function `VirtAddr::as_ptr()` does not take generic arguments.

![image](https://github.com/rcore-os/arceos/assets/36527804/cb9d2b69-1aa7-4403-9830-576c891dffd4)

Reply:
Sorry, please ignore this.